### PR TITLE
[pdata] Rename ValueTypeArray to ValueTypeSlice

### PR DESCRIPTION
**Pull request by dmitryax**
`pdata.ValueTypeArray` is removed without deprecation because it was introduced few commits before and not released yet.

Closes: https://github.com/open-telemetry/opentelemetry-collector/issues/5065

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

File: model/internal/pdata/common_test.go [hunk 1]
```diff
@@ -976,17 +976,17 @@ func generateTestBytesMap() Map {
 	})
 }
 
-func TestAttributeValueArray(t *testing.T) {
-	a1 := NewValueArray()
-	assert.EqualValues(t, ValueTypeArray, a1.Type())
+func TestAttributeValueSlice(t *testing.T) {
```
Comment (bogdandrutu) [inline model/internal/pdata/common_test.go:979]: Separate PR consider to remove `Attribute` from test names.

**Review by bogdandrutu — APPROVED** [review]

File: model/internal/pdata/common_test.go (hunk 1 above)
Comment (dmitryax) [reply model/internal/pdata/common_test.go:979]: Sounds good. I'll do

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by dmitryax** [comment]
done

**Comment by dmitryax** [comment]
_contrib-tests / contrib_tests_ is failing because of removed `ValueTypeArray` which is expected. `ValueTypeArray` isn't part of to the latest release, it was added few after 0.47.0, so I believe it can be removed right away. Let me know if there are any objections.

cc @bogdandrutu @Aneurysm9

**Review by Aneurysm9 — APPROVED** [review]

**Comment by Aneurysm9** [comment]
> `ValueTypeArray` isn't part of to the latest release, it was added after 0.47.0, so I believe it can be removed right away.

I think that's correct.  We don't have any releases with that identifier so there is no need to perpetuate it for compatibility.

**Pull request merged**